service/src/main/java/greencity/service/EventServiceImpl.java

```diff
@@ -733,19 +733,22 @@ private List<Event> getSavedEvents(List<Event> events, Long userId) {
 
     private List<Event> getSocialEvents(List<Event> events) {
         return events.stream().filter(event -> event.getTags().stream()
-            .allMatch(tag -> tag.getType().equals(TagType.EVENT) && tag.getId() == 12L))
+            .anyMatch(tag -> tag.getTagTranslations().stream()
+                .anyMatch(tagTranslation -> tagTranslation.getName().equalsIgnoreCase(SOCIAL_TAG))))
             .collect(Collectors.toList());
     }
 
     private List<Event> getEnvironmentalEvents(List<Event> events) {
         return events.stream().filter(event -> event.getTags().stream()
-            .allMatch(tag -> tag.getType().equals(TagType.EVENT) && tag.getId() == 13L))
+            .anyMatch(tag -> tag.getTagTranslations().stream()
+                .anyMatch(tagTranslation -> tagTranslation.getName().equalsIgnoreCase(ENVIRONMENTAL_TAG))))
             .collect(Collectors.toList());
     }
 
     private List<Event> getEconomicEvents(List<Event> events) {
         return events.stream().filter(event -> event.getTags().stream()
-            .allMatch(tag -> tag.getType().equals(TagType.EVENT) && tag.getId() == 14L))
+            .anyMatch(tag -> tag.getTagTranslations().stream()
+                .anyMatch(tagTranslation -> tagTranslation.getName().equalsIgnoreCase(ECONOMIC_TAG))))
             .collect(Collectors.toList());
     }
```

Comment:
private List<Events getEventsByTagName(final List<Events> events, final String tag) {
return events.stream().filter(event -> event.getTags().stream()
                .flatMap(tag -> tag.getTagTranslations().stream())
                .anyMatch(tagTranslation -> tag.equalsIgnoreCase(tagTranslation.getName())))
            .collect(Collectors.toList());
}

Reply:
Also as a suggestion, this could be part of utility class, eg: EventsUtils